mmcls/evaluation/metrics/single_label.py

```diff
@@ -136,6 +140,18 @@ def __init__(self,
         else:
             self.thrs = tuple(thrs)
 
+        # generate label mask
+        if ann_file_path is not None:
+            with open(ann_file_path, 'r') as f:
+                labels = [
+                    int(item.strip().split()[-1]) for item in f.readlines()
+                ]
+            label_dict = {label: 1 for label in labels}
+            self.label_mask = torch.tensor(
+                [1 if i in label_dict else 0 for i in range(1000)]).cpu()
+        else:
+            self.label_mask = torch.tensor([1] * 1000).cpu()
```

Comment:
The hard-coded 1000 is a little wried because it's only for ImageNet variants. Better to save available categories indices.